### PR TITLE
Add get_all_mods() function + clean up and reformat action_util files

### PR DESCRIPTION
**Pull request by vomindoraan**
## Description

Add a `get_all_mods()` function to core that can be used to easily retrieve the current state of all mods (essentially `real_mods | weak_mods | macro_mods | oneshot_mods`).

I extracted this behavior from `send_keyboard_report()` into a separate function. As a side note, this is why `IS_COMMAND` and other things used to use `keyboard_report->mods` instead of `get_mods()`: the former contained the actual combined state of all mods, whereas the latter only returned real mods.

At some point after this PR, most or all remaining uses of `keyboard_report->mods` should probably be replaced with calls to `get_all_mods()` in order to avoid issues with the report format being different between 6KRO and NKRO, as described in #4838.

I also cleaned up the `action_util.*` files a bit. Namely, I removed trailing whitespace, added blank lines for readability, ensured 2-space indentation is used consistently, added missing parenthses around macro parameters, and so on.

### To-do
- [ ] Add `get_all_mods()` documentation.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by vomindoraan** [comment]
@drashna Since you were mentioning it a couple of weeks ago — I thought about using `get_all_mods()` instead of `get_mods()` in `IS_COMMAND()`, but I decided against it. The reason is that it works fine as is, and changing it would mean changing _all_ `IS_COMMAND()` definitions again, which I really don't want to go through (and I bet you don't either 😄).

File: tmk_core/common/action_util.c [hunk 1]
```diff
@@ -14,26 +14,25 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
+#include "action_util.h"
 #include "host.h"
-#include "report.h"
```
Comment (vomindoraan) [inline tmk_core/common/action_util.c]: `report.h` is already included in `action_util.h`

**Comment by ezuk** [comment]
When you add the docs, I'd love to see a usage example with the reasoning. I.e, not just what it does, but why would I use it in a layout.

**Comment by drashna** [comment]
Well, it would simply some of my code. I use One Shot Mods heavily, for instance. and I do check the mod status to be visually indicated. so this would simplify my code, for instance.

also, even though it would be a pain, get_all_mods may be better for COMMAND, though that could be debated.

**Comment by ezuk** [comment]
that's awesome — I'm just suggesting this rationale be presented in the docs. ;) I guess I should've waited to read those docs, I'm sure they'd be great.

**Comment by drashna** [comment]
Yeah, I agree on both counts.  :)

There are a number of things that need some love in the docs...

**Comment by vomindoraan** [comment]
I will do my best to provide compelling examples when I add it to the docs. Having an indicator for mod state is certainly a good one; I'll be sure to add that. Speaking of the docs, I realized that the regular `get_mods()` isn't actually represented either, so I will be covering that as well.

@drashna If I recall correctly, `get_mods()` works fine with one shot mods for Command, as long as at least one of the Shift keys is being held down (the other can be one-shot). Although I'm not a heavy one shot keys user, I think this is fine. Commands should, imo, always take conscious effort to trigger so it doesn't happen by accident (this is one of the reasons why I dislike Bootmagic so much).

**Comment by vomindoraan** [comment]
@drashna Which doc file would be the best place to document `get_mods()`, `get_all_mods()` etc?

**Comment by drashna** [comment]
> @drashna Which doc file would be the best place to document `get_mods()`, `get_all_mods()` etc?

I think it would be good to cover this in both the feature_macro page, and the useful functions page (and including `get_mods`, and `get_oneshot_mods`).

File: tmk_core/common/action_util.c [hunk 2]
```diff
@@ -50,28 +49,31 @@ extern inline void clear_keys(void);
...
 static uint16_t oneshot_time = 0;
+
 bool has_oneshot_mods_timed_out(void) {
   return TIMER_DIFF_16(timer_read(), oneshot_time) >= ONESHOT_TIMEOUT;
 }
 #else
-bool has_oneshot_mods_timed_out(void) {
-    return false;
-}
+bool has_oneshot_mods_timed_out(void) { return false; }
```
[the remark this answers is not in the document]
Comment (vomindoraan) [reply tmk_core/common/action_util.c]: I'm not a fan of it either, but at the time I did it to be consistent with the rest of the file, as most of the one-liners are already in a collapsed form like this. Let me know if you would still like me to revert it.

**Comment by vomindoraan** [comment]
Sure thing, I will revert the indentation changes. The added/removed newlines should be fine, though, as they aren't affecting any lines with code?

**Comment by drashna** [comment]
There are some merge conflicts that have cropped up? 

Could the PR be updated to address these?

**Pull request closed without merging**